frontend/src/components/FilePreview/index.tsx

```diff
@@ -0,0 +1,127 @@
+  const { network } = useNetwork();
+  const [isFilePreview, setIsFilePreview] = useState(false);
+  const [loading, setLoading] = useState(true);
+  const [error, setError] = useState<string | null>(null);
+  const [isDecrypted, setIsDecrypted] = useState(false);
+  const [decryptionError, setDecryptionError] = useState<string | null>(null);
+  const [file, setFile] = useState<Blob | null>(null);
+  const [textContent, setTextContent] = useState<string | null>(null);
+
+  const networkId = useMemo(() => {
+    switch (network.id) {
+      case AutoDriveNetworkId.LOCAL:
+      case AutoDriveNetworkId.MAINNET:
+        return AutoUtilsNetworkId.MAINNET;
+      case AutoDriveNetworkId.TAURUS:
+        return AutoUtilsNetworkId.TAURUS;
+      default:
+        return AutoUtilsNetworkId.MAINNET;
+    }
+  }, [network]);
+
+  const gatewayUrl = EXTERNAL_ROUTES.gatewayObjectDownload(metadata.dataCid);
+
+  const fetchFile = useCallback(
+    async (password?: string) => {
+      // If file is encrypted and no password provided, don't fetch
+      if (metadata.uploadOptions?.encryption && !password && !isDecrypted) {
+        setIsFilePreview(false);
+        setLoading(false);
+        return;
+      }
+
+      // For non-encrypted files that can be displayed directly,
+      if (!metadata.uploadOptions?.encryption && canDisplayDirectly(metadata)) {
+        setIsFilePreview(true);
+        setFile(null);
+        setLoading(false);
+        return;
+      }
+
+      // Encrypted files always need to be fetched and decrypted
+      setIsFilePreview(true);
+      setLoading(true);
+
+      try {
+        const response = await fetch(gatewayUrl);
+        if (!response.ok) {
+          throw new Error(
+            `Failed to fetch file: ${response.status} ${response.statusText}`,
+          );
+        }
+        const blob = await response.blob();
+        setFile(blob);
+        // For text-based files, also read the content
+        if (needsContentParsing(metadata)) {
+          const text = await blob.text();
+          setTextContent(text);
+        }
+        // Handle decryption if needed
+        if (metadata.uploadOptions?.encryption && password) {
+          try {
+            const encryptedFileData = {
+              dataArrayBuffer: await blob.arrayBuffer(),
+              name: metadata.name ?? '',
+              rawData: '',
+              uploadOptions: metadata.uploadOptions,
+              isEncrypted: true,
+            };
+            const decryptedFileData = await decryptFileData(
+              password,
+              encryptedFileData,
+            );
+            const decryptedBlob = await processFileData(decryptedFileData);
+            setFile(decryptedBlob);
+            setIsDecrypted(true);
+            setLoading(false);
+            return;
+          } catch {
+            setDecryptionError('Invalid password or decryption failed');
+          }
+        }
```

Comment:
We should clear the password from state after it is used. We should also think about how we want the lifecycle of decrypted data to be handled.

Reply:
Password isn't saved on the state, is just a callback parameter: [see here](https://github.com/autonomys/auto-drive/pull/425/files#diff-255c318571563f19b9c163ed93b9823c0ac9a1f33dfa22dc7b43585b757e4c0bR40)